clang/include/clang/Basic/Attr.td

```diff
@@ -1303,7 +1303,7 @@ def SYCLIntelMaxGlobalWorkDim : InheritableAttr {
 def SYCLIntelNoGlobalWorkOffset : InheritableAttr {
   let Spellings = [CXX11<"intelfpga","no_global_work_offset">,
                    CXX11<"intel","no_global_work_offset">];
-  let Args = [BoolArgument<"Enabled", 1>];
+  let Args = [ExprArgument<"Value", /*optional*/1>];
   let LangOpts = [SYCLIsDevice, SYCLIsHost];
```

Comment:
I would expect there to be a `let TemplateDependent = 1;` here to instantiate the attribute within a template. But then again, we're following the template instantiation pattern used by other attributes (which makes me wonder if someday I should go add some more tablegen support for this stuff).

I'd say for right now this is fine as-is, I guess.

Reply:
I will leave this now. Inside Attr.td file, i see that we used let TemplateDependent = 1 only for GNU/Clang attribute spellings. Since we are  following the template instantiation pattern used by other SYCL attributes here, i will not add this now.  We can add this later if we decide to go with it.

Reply:
Agreed!